pkg/controller/job/job_controller.go

```diff
@@ -1391,8 +1424,79 @@ func getStatus(job *batch.Job, pods []*v1.Pod, uncounted *uncountedTerminatedPod
 		} else {
 			return isPodFailed(p, job, uncounted != nil)
 		}
-	}))
-	return succeeded, failed
+	})
+	failed += int32(len(failedPods))
+
+	finishedTimeSortFunction := func(p1, p2 *v1.Pod) bool {
+		p1FinishTime := getFinishedTime(p1)
+		p2FinishTime := getFinishedTime(p2)
+
+		return p1FinishTime.Before(&p2FinishTime)
+	}
+
+	sort.Slice(succeededPods, func(i, j int) bool {
+		return finishedTimeSortFunction(succeededPods[i], succeededPods[j])
+	})
+
+	sort.Slice(failedPods, func(i, j int) bool {
+		return finishedTimeSortFunction(failedPods[i], failedPods[j])
+	})
+
+	return succeeded, failed, succeededPods, failedPods
+}
+
+func getFinishedTime(p *v1.Pod) metav1.Time {
+	finishTime := p.Status.Conditions[0].LastTransitionTime
+	for _, cond := range p.Status.Conditions {
+		if finishTime.Before(&cond.LastTransitionTime) {
+			finishTime = cond.LastTransitionTime
+		}
+	}
+
+	return finishTime
+}
+
+func computeNewBackoff(oldBackoff int32, succeededPods, failedPods []*v1.Pod) int32 {
```

Comment:
I would like this to be moved to `backoff_utils.go`. Also, it seems to could be easier to follow if we just computed `backoffCount` as the number of pods failed since the last success. It could be then translated to `backoff`. Analogously, I would suggest to keep `backoffCount` in-memory cache instead of the number of seconds.

Reply:
Done